### PR TITLE
Material UI Checkbox fix required from schema

### PR DESCRIPTION
**Pull request by stanlemon**
### Reasons for making this change

Addresses a bug in #1796 for the checkbox widget when it is required according to the schema.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [X] **I'm adding or updating code**
  - [X] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/material-ui/src/CheckboxWidget/CheckboxWidget.tsx [hunk 1]
```diff
@@ -19,6 +22,11 @@ const CheckboxWidget = (props: WidgetProps) => {
     onFocus,
   } = props;
 
+  // Because an unchecked checkbox will cause html5 validation to fail, only add
+  // the "required" attribute if the field value must be "true", due to the
+  // "const" or "enum" keywords
+  const required = schemaRequiresTrueValue(schema);
```
Comment (epicfaace) [inline packages/material-ui/src/CheckboxWidget/CheckboxWidget.tsx:28]: If HTML5 validation is turned off, should we still use `schemaRequiresTrueValue(schema)` or should we revert to `props.required`?

Comment (stanlemon) [reply packages/material-ui/src/CheckboxWidget/CheckboxWidget.tsx:28]: That's a fair question and I don't know the answer honestly. This change in particular brings the behavior into alignment with the bootstrap theme (https://github.com/rjsf-team/react-jsonschema-form/blob/master/packages/core/src/components/widgets/CheckboxWidget.js#L51).  I'd defer to your direction on the question though, so just let me know if you'd like for me to make changes here.

Comment (epicfaace) [reply packages/material-ui/src/CheckboxWidget/CheckboxWidget.tsx:28]: Ah, makes sense, let's just keep it the way you did it then.

File: packages/material-ui/package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
   "scripts": {
     "start": "tsdx watch",
     "build": "rimraf dist && tsdx build",
-    "test": "tsdx test"
+    "test": "tsdx test --updateSnapshot"
```
Comment (epicfaace) [inline packages/material-ui/package.json:14]: I think you should revert this (so CI doesn't update snapshots too)

Comment (stanlemon) [reply packages/material-ui/package.json]: Good catch, didn’t mean to let that sneak in. Thanks!

**Comment by stanlemon** [comment]
@epicfaace Bump ☺️

**Pull request merged**